src/mixins/limit-in-parallel.ts

```diff
@@ -1,26 +1,43 @@
 import _ from 'lodash';
 
-// eslint-disable-next-line firecloud/underscore-prefix-non-exported
+import {
+  Fn,
+  FnChangeReturnType,
+  Unpromise
+} from '../types';
+
+// eslint-disable-next-line firecloud/underscore-prefix-non-exported, @typescript-eslint/explicit-function-return-type
 let AsyncFunction = Object.getPrototypeOf(async function() { /* noop */ }).constructor;
 
 // useful for checks like `err instanceof _.LimitInParallelError`
 export class LimitInParallelError extends Error {
 }
 
+export interface LimitInParallelOptions {
+  limit?: number,
+  throwErr?: boolean
+}
+
 // TODO is there a way not to duplicate the function just because origFn is sync or async?
 
 /**
  * Part of `lodash-firecloud`.
  *
  * Create a function that can only run maximum n times in parallel.
  *
- * @param {Function} origFn The function to have its parallelism limited.
- * @param {Object} options The options.
- * @param {number} [options.limit=1] Specifies the maximum number of parallel executions.
- * @param {boolean} [options.throwErr=false] Specifies whether to throw an error when the limit has been reached.
- * @returns {Function} Returns the new limited function.
+ * @param origFn The function to have its parallelism limited.
+ * @param options The options.
+ * @param [options.limit=1] Specifies the maximum number of parallel executions.
+ * @param [options.throwErr=false] Specifies whether to throw an error when the limit has been reached.
+ * @returns Returns the new limited function.
  */
-export let limitInParallel = function(origFn, options = {}) {
+// eslint-disable-next-line @typescript-eslint/promise-function-async, @typescript-eslint/explicit-function-return-type
+export let limitInParallel = function<
```

Comment:
I think line 59-73 can be simplified to:
```ts
      try {
        return await origFn(...args);
      } finally {
        activeCount = activeCount - 1;
      }	
```

Or?

Reply:
Just a thought: If I have function foo that returns a promise and I wrap it using `limitInParallel` the behavior in case of a rejected promise will change.

If the promise returned by foo rejects, blabla() will get called in this case:

```ts
foo().catch(blabla());
```

But not in this case:
```ts
limitInParallel(foo)().catch(blabla());
```

Makes sense?